packages/openlayers/lib/map/create-map.ts

```diff
@@ -30,11 +30,16 @@ import {
   WmtsEndpoint,
 } from "@camptocamp/ogc-client";
 import { MapboxVectorLayer } from "ol-mapbox-style";
+import { Tile } from "ol";
+import { getEndpoint, tileLoadErrorCatchFunction } from "./handle-errors";
 
 const GEOJSON = new GeoJSON();
 const WFS_MAX_FEATURES = 10000;
 
-export async function createLayer(layerModel: MapContextLayer): Promise<Layer> {
+export async function createLayer(
+  layerModel: MapContextLayer,
+  map: Map,
```

Comment:
I think this is too big of an impact for this change, especially since we're just using this as a "hack" to push the event upwards

Reply:
Hum, I'm not sure if I entirely understand the impact to pass the map here. Does this create too many references to the map for example? Could you explain this a little more?

Reply:
This function is part of the API of the SDK; requiring a map to create a layer changes a lot of things, as you could see when you had to change all the tests. Being able to create a layer without a map makes the API easier and more intuitive to use. It is after all the case on OL: we create a source, then a layer, then add a layer to the map

Reply:
Ah right, didn't think about the API exposition